test/packages_test.rb

```diff
@@ -836,4 +847,39 @@ def pattern(properties = {})
       end
     end
   end
+
+  describe "#Reset" do
+    # Reset package changes done by YaST but select the installe products back.
+    # Do not select previously not selected products which were added by
+    # registration (see bsc#963036).
+    it "does not select previously unselected items" do
+      allow(Yast::Pkg).to receive(:PkgApplReset)
+
+      allow(Yast::Pkg).to receive(:ResolvableProperties).and_return(
+        [product("name" => "p1", "status" => :selected), product("name" => "p2")]
+      )
+
+      expect(Yast::Pkg).to receive(:ResolvableInstall).with("p1", :product)
+      expect(Yast::Pkg).not_to receive(:ResolvableInstall).with("p2", :product)
+
+      Yast::Packages.Reset([:product])
+    end
+
+    # When restoring the selected products after reset ignore the products
```

Comment:
Captial R in Reset?

missing a comma after reset

the products -> all products